src/OrchardCore/OrchardCore.Media.Core/OrchardCore.Media.Core/DefaultImageStreamService.cs

```diff
@@ -0,0 +1,33 @@
+using System.IO;
+
+namespace OrchardCore.Media.Core
+{
+    public class DefaultImageStreamService : IImageStreamService
+    {
+        private readonly IImageService _imageService;
+        public DefaultImageStreamService(IImageService imageService)
```

Comment:
I think this will fail to resolve from DI if there is no implementation of `IImageService` registered.

I'm guessing the intention here is to allow a developer to transform an image during upload, so for example if a user uploaded a 10mb image file, it would be transformed down to a 1MB file automatically.

Am I correct in that assumption?

It seems like a good idea, given that we know from issues such as https://github.com/OrchardCMS/OrchardCore/issues/4827 that very large images can cause memory issues.

What I think would be better is to implement an event handler interface like `IMediaFileCreated`.

This would be used in the `MediaFileStore`

some psuedo code...
injected into it's constructor like all the other event handlers are
`IEnumerable<IMediaFileCreated>`

and in the `MediaFileStore`
```
foreach(var mediaFileCreatedEvent in mediaFileCreatedEvents){
   await mediaFileCreatedEvent(stream)
}
```

We use event handlers like this in a lot of places in OC, and it would be a lot less code, plus would be acting in the correct place.

Reply:
@deanmarcussen Thanks for the comment , I will try to go your way. please make this PR as draft.

Reply:
@infofromca cool, just make sure to pass the filename into the event as well, so that it can determine if it's a resizable type of image/file :)